cmd/temporal-aws-sdk-gen/internal/parser.go

```diff
@@ -14,31 +19,45 @@ var blacklistedMethods = map[string]string{
 	"Lambda": "InvokeAsync",
 }
 
-type AWSSDKParser struct {
-	SdkDirectory string
-}
-
 // ParseAwsSdk calls generator function for each AWS Service iDefinition
-func (p *AWSSDKParser) ParseAwsSdk(serviceName string) ([]*InterfaceDefinition, error) {
-	var serviceNames []string
+func ParseAwsSdk(serviceName string) ([]*InterfaceDefinition, error) {
+	var pkgs []*packages.Package
+
+	config := &packages.Config{
+		Mode: packages.NeedName |
+			packages.NeedFiles |
+			packages.NeedCompiledGoFiles |
+			packages.NeedImports |
+			packages.NeedTypes |
+			packages.NeedTypesSizes |
+			packages.NeedTypesInfo,
+	}
+
 	if serviceName == "" {
-		serviceDir := p.SdkDirectory + "/service"
-		files, err := ioutil.ReadDir(serviceDir)
+		allPkgs, err := packages.Load(config, "github.com/aws/aws-sdk-go/service/...")
```

Comment:
Let's move the repository path into a variable.
Also, need to be able to specify branch/sha when invoking the generator.

Reply:
I'm OK to add this feature in a separate PR.

Reply:
I think branch/sha should be the same as the version in the module the generator runs. But let's discuss it in a different PR/issue.

Reply:
I missed that it already uses the version from the module.